service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProvider.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.fpl.service.email.content;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+
+@Service
+public class C2UploadedEmailContentProvider extends AbstractEmailContentProvider {
+
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    protected C2UploadedEmailContentProvider(@Value("${ccd.ui.base.url}") String uiBaseUrl,
+                                             ObjectMapper objectMapper) {
+        super(uiBaseUrl);
+        this.objectMapper = objectMapper;
+    }
+
+    public Map<String, Object> buildC2UploadNotification(final CaseDetails caseDetails) {
+        CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+        final String subjectLine = EmailNotificationHelper.buildSubjectLine(caseData);
+        return Map.of(
+            "subjectLine", subjectLine,
+            "hearingDetailsCallout", subjectLine,
```

Comment:
yes it is. Had to separate as we tried to decide if (or not) to add _hearingDate_ to _ hearingDetailsCallout_.

Reply:
so why are we using two variables? can we not use subject line twice in the template?

Reply:
marked as resolved but not resolved @Gogetter

Reply:
Ok I think only one variable now so it is probably fine to resolve.

Reply:
@swalker125, there is currently an ongoing discussion as to what the callout should be (with or without the hearing date). Left this in as at the moment its without the hearingDate, but this could change and `hearingDetailsCallout` value will be different from `subjectLine`